lib/SIL/Verifier/MemoryLifetime.cpp

```diff
@@ -745,6 +767,37 @@ void MemoryLifetimeVerifier::setBitsOfPredecessor(Bits &bits,
   if (!pred)
     return;
 
+  if (auto *ccabi =
+          dyn_cast<CheckedCastAddrBranchInst>(pred->getTerminator())) {
+    if (block == ccabi->getSuccessBB()) {
+      switch (ccabi->getConsumptionKind()) {
+      case CastConsumptionKind::BorrowAlways:
+        llvm_unreachable("Only for SILGenPattern");
+      case CastConsumptionKind::CopyOnSuccess:
+        locations.setBits(bits, ccabi->getDest());
+        break;
+      case CastConsumptionKind::TakeOnSuccess:
+      case CastConsumptionKind::TakeAlways:
+        locations.clearBits(bits, ccabi->getSrc());
```

Comment:
Unfortunately clearBits only works in case setBitsOfPredecessor is called from checkBlock.
When initializing the data flow, the "bits" argument to setBitsOfPredecessor is the gen-set. This works for try_apply, because only gen-bits are added, but not for cast_addr_br, where bits are "killed".

What you can do is to pass 2 closures to setBitsOfPredecessor: one for setting and one for clearing bits.
In case of initDataflowInBlock, the closures would call state.genBits and state.killBits, in case of checkBlock, the closures would call locations.setBits and locations.clearBits.